app/code/Magento/Catalog/Controller/Adminhtml/Product/Set/Edit.php

```diff
@@ -6,53 +6,76 @@
  */
 namespace Magento\Catalog\Controller\Adminhtml\Product\Set;
 
-use Magento\Framework\App\Action\HttpGetActionInterface as HttpGetActionInterface;
+use Magento\Framework\Registry;
+use Magento\Backend\App\Action\Context;
+use Magento\Backend\Model\View\Result\Page;
+use Magento\Framework\View\Result\PageFactory;
+use Magento\Framework\Controller\ResultFactory;
+use Magento\Framework\Controller\ResultInterface;
+use Magento\Eav\Model\Entity\Attribute\SetFactory;
+use Magento\Catalog\Controller\Adminhtml\Product\Set;
+use Magento\Framework\App\Action\HttpGetActionInterface;
 
-class Edit extends \Magento\Catalog\Controller\Adminhtml\Product\Set implements HttpGetActionInterface
+/**
+ * Class Edit
+ *
+ * @package Magento\Catalog\Controller\Adminhtml\Product\Set
+ */
+class Edit extends Set implements HttpGetActionInterface
 {
     /**
-     * @var \Magento\Framework\View\Result\PageFactory
+     * @var PageFactory
      */
     protected $resultPageFactory;
 
     /**
-     * @param \Magento\Backend\App\Action\Context $context
-     * @param \Magento\Framework\Registry $coreRegistry
-     * @param \Magento\Framework\View\Result\PageFactory $resultPageFactory
+     * @var SetFactory
+     */
+    protected $setFactory;
+
+    /**
+     * Edit constructor.
+     *
+     * @param Context $context
+     * @param Registry $coreRegistry
+     * @param PageFactory $resultPageFactory
+     * @param SetFactory $setFactory
      */
     public function __construct(
-        \Magento\Backend\App\Action\Context $context,
-        \Magento\Framework\Registry $coreRegistry,
-        \Magento\Framework\View\Result\PageFactory $resultPageFactory
+        Context $context,
+        Registry $coreRegistry,
+        PageFactory $resultPageFactory,
+        SetFactory $setFactory
     ) {
         parent::__construct($context, $coreRegistry);
         $this->resultPageFactory = $resultPageFactory;
+        $this->setFactory = $setFactory;
     }
 
     /**
-     * @return \Magento\Backend\Model\View\Result\Page
+     * Controller execute method
+     *
+     * @return ResultInterface
      */
-    public function execute()
+    public function execute(): ResultInterface
```

Comment:
Unfortunately, we are not allowed to change public method signature according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

Reply:
Done, thank you!